### DISCUSSION TITLE
Tooltip 如何调整宽度？

### DISCUSSION THREAD

**Discussion by Song-aff** [ant-design/ant-design#37464] [Q&A]
Tooltip在高度不够时，会调整位置，为什么不通过增加宽度的方式来展示更多内容？
![image](https://user-images.githubusercontent.com/24764561/189058453-0422cf24-f710-4fb1-bb5a-e755ad914ff9.png)

**Comment by Song-aff** [comment]
没有人有相同问题吗

**Reply by fnoopv** [reply]
没有，感觉一下，在tooltip内容足够多时是一个特别长的细长条看着舒服还是一块看着舒服？当然截图里面的调整位置可能是BUG

**Reply by Song-aff** [reply]
这样内容过多的时候，不也成了竖着的一长条了，稍微增加些宽度，才能更像一块。我理解可能是会对定位产生麻烦

**Reply by fnoopv** [reply]
这取决于内容过多的“多”是怎么定义的，像你例子里面的多可能适合调整宽度来达到更好的显示结果，但是假如是你例子里面内容的几倍或是几十倍这么多，调整宽度也没用，因为你不能确定内容的长短，也就没办法去设置一个合适的宽度。所以根本解决方案是控制`tooltip`内容的长度.

**Reply by Song-aff** [reply]
我也这么觉得，但是业务不这样认为 😂

**Reply by fnoopv** [reply]
刚去看了一下`Tooltip`的属性，`overlayStyle`可以设置`Tooltip`卡片的宽度，但是最大宽度取决于父容器的宽度。所以可以用`overlayInnerStyle`来设置`Tooltip`内容的宽度，这个宽度可以超出父级容器的宽度，比如下面这种效果
![image](https://user-images.githubusercontent.com/24599198/191410798-1f1c9d41-5ee8-492c-978a-9de2bb9fc764.png)
代码:
![image](https://user-images.githubusercontent.com/24599198/191410837-0e4c6ca7-7d79-4ada-9c2f-2485f3ac3c2e.png)

**Reply by Song-aff** [reply]
wow,我去试试
